server/tests/user.integration.spec.js

```diff
@@ -45,51 +56,88 @@ describe('Notifications API:', () => {
             .post(`${baseURL}/users`)
-            .send(testUserObject)
+            .send(userA)
             .expect(httpStatus.UNAUTHORIZED)
         );
 
         it('Should fail with badtoken in Authorization header', () => request(app)
             .post(`${baseURL}/users`)
             .set('Authorization', 'Bearer badtoken')
-            .send(testUserObject)
+            .send(userA)
             .expect(httpStatus.UNAUTHORIZED)
         );
+    });
+
+    describe('Create User and Update Device Token', () => {
+        it('should create and respond with a user', () => request(app)
+            .post(`${baseURL}/users`)
+            .set('Authorization', `Bearer ${auth}`)
+            .send(userA)
+            .expect(httpStatus.OK)
+            // eslint-disable-next-line no-underscore-dangle
+            .then(res => User.findById(res.body.user._id))
+            .then((user) => {
+                // eslint-disable-next-line no-unused-expressions
+                expect(user).to.not.be.null;
+                expect(user.username).to.equal(userA.username);
+                expect(user.uuid).to.equal(userA.uuid);
+            })
+        );
 
-        it('should should create and respond with a user', () => request(app)
+        it('should return success but not create a duplicate user', () => request(app)
             .post(`${baseURL}/users`)
             .set('Authorization', `Bearer ${auth}`)
-            .send(testUserObject)
+            .send(userA)
             .expect(httpStatus.OK)
-            .then((res) => {
-                User.findById(res.body.user.id)
-                  .then((user) => {
-                      expect(user.username).to.equal(testUserObject.username);
-                  });
+            // eslint-disable-next-line no-underscore-dangle
+            .then(res => User.find({ username: res.body.user.username }))
+            .then((users) => {
+                // eslint-disable-next-line no-unused-expressions
+                expect(users).to.not.be.null;
+                expect(users.length).to.equal(1);
             })
         );
 
         it('should create a device for a user', () => request(app)
             .post(`${baseURL}/users/updateDevice`)
             .set('Authorization', `Bearer ${auth}`)
-            .send(deviceRequestData)
+            .send(userADeviceA)
             .expect(httpStatus.OK)
-            .then(res => User.findOne({   // eslint-disable-line no-unused-vars
-                where: {
-                    username: testUserObject.username,
-                },
-                include: [{
-                    model: Device,
-                }],
+            .then(() => User.findOne({
+                username: userA.username,
+            }))
+            .then((user) => {
+                // eslint-disable-next-line max-len
+                const device = user.devices.find(_device => _device.token === userADeviceA.token);
+                expect(device).to.not.equal(null);
```

Comment:
This could be added, correct? `expect(device.type).to.equal('iOS')`

Reply:
Totally